WcaOnRails/app/models/team.rb

```diff
@@ -18,4 +18,23 @@ def membership_periods_cannot_overlap_for_single_user
       end
     end
   end
+
+  validate :member_cannot_be_empty
+  def member_cannot_be_empty
+    team_members.each do |member|
+      if member.user == nil
+        errors[:base] << "Team member cannot be empty"
+      end
+    end
+  end
+
+  validate :start_date_cannot_be_empty
+  def start_date_cannot_be_empty
+    team_members.each do |member|
+      if member.start_date == nil
+        errors[:base] << "Membership start cannot be empty"
+      end
+    end
+  end
+
```

Comment:
I would rather suggest something like this in the team_meber.rb
```ruby
validates :start_date, presence: true
validates :user, presence: true
```

Note: if you're wondering why there's such a iterating validation above yours, it's because we accept nested atrributes for team members so on update action in teams controller, team members' attributes are asigned but they are not saved before validations. So we cannot grab the other team members in one of them in order to validate whether their membership periods don't overlap.